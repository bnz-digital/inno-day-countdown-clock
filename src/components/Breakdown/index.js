import React from 'react'
import { number, string } from 'prop-types'

import StyledBreakdown from './styled'

export default function Breakdown ({ days, hrs, mins, secs, unstyled }) {
  const clock = days > 0
    ? `${days}:${hrs}:${mins}:${secs}`
    : `${hrs}:${mins}:${secs}`

  return unstyled
    ? <span>{clock}</span>
    : <StyledBreakdown>{clock}</StyledBreakdown>
}

Breakdown.propTypes = {
  days: number,
  hrs: string.isRequired,
  mins: string.isRequired,
  secs: string.isRequired
}
