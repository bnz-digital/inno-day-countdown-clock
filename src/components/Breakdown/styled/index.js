import styled from 'styled-components'

const StyledBreakdown = styled.footer`
  color: #FAA61B;
  font-size: 4vw;
  line-height: 1.5;
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  padding: 1vw;
  text-align: center;
`
StyledBreakdown.displayName = 'StyledBreakdown'

export default StyledBreakdown
