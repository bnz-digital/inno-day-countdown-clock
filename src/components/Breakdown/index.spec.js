import React from 'react'
import { shallow } from 'enzyme'

import Breakdown from '.'

describe('components:Breakdown', () => {
  it('renders <Breakdown/> correctly with days', () => {
    expect(
      toJson(shallow(<Breakdown days={5} hrs='13' mins='05' secs='51' />))
    ).toMatchSnapshot()
  })

  it('renders <Breakdown/> correctly without days', () => {
    expect(
      toJson(shallow(<Breakdown hrs='13' mins='05' secs='51' />))
    ).toMatchSnapshot()
  })

  it('renders <Breakdown/> correctly without styling', () => {
    expect(
      toJson(shallow(<Breakdown hrs='13' mins='05' secs='51' unstyled />))
    ).toMatchSnapshot()
  })
})
