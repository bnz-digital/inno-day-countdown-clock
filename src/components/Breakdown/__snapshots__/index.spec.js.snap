// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components:Breakdown renders <Breakdown/> correctly with days 1`] = `
<StyledBreakdown>
  5:13:05:51
</StyledBreakdown>
`;

exports[`components:Breakdown renders <Breakdown/> correctly without days 1`] = `
<StyledBreakdown>
  13:05:51
</StyledBreakdown>
`;

exports[`components:Breakdown renders <Breakdown/> correctly without styling 1`] = `
<span>
  13:05:51
</span>
`;
